homeassistant/components/device_tracker/icloud.py

```diff
@@ -5,17 +5,24 @@
 https://home-assistant.io/components/device_tracker.icloud/
 """
 import logging
-import re
+import voluptuous as vol
 
-from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.const import (CONF_PASSWORD, CONF_USERNAME,
+                                 EVENT_HOMEASSISTANT_START)
 from homeassistant.helpers.event import track_utc_time_change
+from homeassistant.util import slugify
 
 _LOGGER = logging.getLogger(__name__)
 
-REQUIREMENTS = ['pyicloud==0.8.3']
+REQUIREMENTS = ['pyicloud==0.9.1']
 
 CONF_INTERVAL = 'interval'
-DEFAULT_INTERVAL = 8
+
+PLATFORM_SCHEMA = vol.Schema({
+    vol.Required(CONF_USERNAME): vol.Coerce(str),
+    vol.Required(CONF_PASSWORD): vol.Coerce(str),
+    vol.Optional(CONF_INTERVAL, default=8): vol.Coerce(int)
+    }, extra=vol.ALLOW_EXTRA)
```

Comment:
device_tracker uses `cv.PLATFORM_SCHEMA` which allows_extra. `known_devices` not in there yet.

Will work on that tomorrow, but this will have a much wider impact then